server/lib/nl/fulfillment/fulfiller.py

```diff
@@ -233,3 +239,50 @@ def _prune_non_country_special_dc_vars(state: PopulateState):
                              list(dropped_vars))
 
   state.chart_vars_map = pruned_chart_vars_map
+
+
+#
+# Prune topic chart vars for toolformer_rig mode
+#
+def _prune_topic_vars_for_rig(state: PopulateState):
+  # Set of all svs that were detected
+  detected_svs = set([
+      sv for sv in state.uttr.detection.svs_detected.single_sv.svs
+      if not sv.startswith(_TOPIC_PREFIX)
+  ])
+  # skip topics if there are any non topic vars detected
```

Comment:
add rationale?

Reply:
hmm, I'm not sure about this.  for example, topics might be at the top and the vars after that might only be poor matches, esp compared to the variables in the t opic.

Reply:
I was going back and forth on this.

One example scenario I was thinking was:
- topic A has variables sv1, sv2 (listed in that order)
- sv1 and sv2 are both separately detected as well
- the order of matching is topicA, sv2, sv1
- if we keep topicA with pruned topics, sv1 will be returned as first chart instead of sv2

& I was thinking for your example, since that's the current behavior, by skipping topics when svs are detected, we won't be doing any worse than currently & slightly better with the addition of topics when no single svs are returned

wdyt (also not sure how likely my example scenario is)

Reply:
Actually how about the following logic instead:

1. Collect all SVs part of the topic chart_vars.
2. Collect all non-topic SVs in chart_vars.
3. Walk down the matched vars list:
    a. If topic, skip
    b. Else, if part of (1) and not part of (2), introduce chart_vars entry.

This keeps the original order of the SVs faithfully.

Reply:
i like this, updated